src/main/java/org/avarion/graves/listener/PlayerInteractAtEntityListener.java

```diff
@@ -32,7 +32,8 @@ public void onPlayerInteractAtEntity(@NotNull PlayerInteractAtEntityEvent event)
             Grave grave = plugin.getEntityDataManager().getGrave(entity);
 
             if (grave != null) {
-                event.setCancelled(plugin.getGraveManager().openGrave(player, entity.getLocation(), grave));
```

Comment:
You mentioned:
> Interacting with entities should be cancelled no matter what happens if a grave is detected.
> If these events are not cancelled, players can take items from item frames, and take the equipment off of armor stands.

Isn't that what was happening here? Only if it's a grave that is opened, the event should be cancelled, no?

However, you already validated with the if-check before that it was an actual grave... So it does make sense to always cancel it.

Could you elaborate a bit what I can do to reproduce this bug so I can understand the situation better?

Reply:
If you'd like to reproduce the bug:
- Set grave to spawn an armor stand that is not a marker, lets say with a chest item
- Make graves protected
- Kill the player in-game, and have another player try to access the grave while it is protected
- The other player will pick up the chest from the armor stand, since opening the grave will return `false` so the entity interact event isn't cancelled